src/hooks/screen/useChatScreenHook.ts

```diff
@@ -27,33 +31,52 @@ interface ScrollContextProps {
 export const ScrollContext = React.createContext<ScrollContextProps | null>(null);
 
 function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
-  const {localDb, chat, refresh} = useLocalDatabaseHook();
-  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook(type);
-  const {anonProfileId} = useUserAuthHook();
+  const {localDb, refresh, otherListener} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen, setChannelAsRead} =
+    useChatUtilsHook(type);
+  const {anonProfileId, signedProfileId} = useUserAuthHook();
+  const {queue} = useDatabaseQueueHook();
 
   const [selfAnonUserInfo, setSelfAnonUserInfo] = React.useState<any>(null);
   const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
   const initChatData = async () => {
     if (!localDb || !selectedChannel) return;
+    if (!otherListener[`chat_${selectedChannel?.id}`]) return;
+
     try {
-      const myUserId = await getUserId();
-      const myAnonymousId = await getAnonymousUserId();
-      const data = (await ChatSchema.getAll(
-        localDb,
-        selectedChannel?.id,
-        myUserId,
-        myAnonymousId
-      )) as ChatSchema[];
-      setChats(data);
+      queue.addPriorityJob({
+        priority: QueueJobPriority.HIGH,
+        operationLabel: DatabaseOperationLabel.ChatScreen_GetChat,
+        id: selectedChannel?.name,
+        task: async () => {
+          setChannelAsRead(selectedChannel, true);
+          const data = (await ChatSchema.getAll(
+            localDb,
+            selectedChannel?.id,
+            signedProfileId,
+            anonProfileId
+          )) as ChatSchema[];
+          setChats(data);
 
-      if (type === 'ANONYMOUS') {
-        const userInfo = await UserSchema.getSelfAnonUserInfo(
-          localDb,
-          anonProfileId,
-          selectedChannel?.id
-        );
+          return data;
+        }
+      });
 
-        setSelfAnonUserInfo(userInfo);
+      if (type === 'ANONYMOUS') {
+        queue.addPriorityJob({
+          priority: QueueJobPriority.MEDIUM,
+          operationLabel: DatabaseOperationLabel.ChatScreen_GetSelfAnonUserInfo,
+          id: selectedChannel?.name,
+          task: async () => {
+            const userInfo = await UserSchema.getSelfAnonUserInfo(
+              localDb,
+              anonProfileId,
+              selectedChannel?.id
+            );
+            setSelfAnonUserInfo(userInfo);
+          }
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `initChatData` function is a significant improvement over the previous one. It uses the queue to manage database operations, which can enhance performance and maintainability. However, there's no error handling for the tasks added to the queue. If an error occurs during the execution of these tasks, it might lead to unexpected behavior or crashes. Consider adding error handling within each task in the queue.
```diff
     try {
      queue.addPriorityJob({
        priority: QueueJobPriority.HIGH,
        operationLabel: DatabaseOperationLabel.ChatScreen_GetChat,
        id: selectedChannel?.name,
        task: async () => {
          setChannelAsRead(selectedChannel, true);
          let data;
          try {
            data = (await ChatSchema.getAll(
              localDb,
              selectedChannel?.id,
              signedProfileId,
              anonProfileId
            )) as ChatSchema[];
            setChats(data);
          } catch (e) {
            console.error('Error while getting all chats:', e);
          }
          return data;
        }
      });
 
      if (type === 'ANONYMOUS') {
        queue.addPriorityJob({
          priority: QueueJobPriority.MEDIUM,
          operationLabel: DatabaseOperationLabel.ChatScreen_GetSelfAnonUserInfo,
          id: selectedChannel?.name,
          task: async () => {
            let userInfo;
            try {
              userInfo = await UserSchema.getSelfAnonUserInfo(
                localDb,
                anonProfileId,
                selectedChannel?.id
              );
              setSelfAnonUserInfo(userInfo);
            } catch (e) {
              console.error('Error while getting self anonymous user info:', e);
            }
          }
        });
      }
    } catch (e) {
      console.log(e, 'error get all chat');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->